enhancement: improve error handling on claim flow (#4488)
iotaledger/firefly: docs/guides/coding-conventions/errors.md

    throw new InvalidStrongholdPasswordError()
}
```

## Errors from wallet.rs

We handle errors from wallet.rs by using the `handleError` function, e.g.

```typescript
try {
    ...
} catch (err) {
    handleError(err)
}
```

> This will automatically check, if the error comes from wallet.rs or not. Step-by-step the handlers for the corresponding errors (e.g. `ClientError`, `InsufficientFunds`, ...) will be added to this function. If you encounter a not-handled error, create the handler and add the type and the corresponding handlers in the following position in the `handleWalletRsError.ts` :

Reviewer: ```suggestion
This automatically checks, if the error comes from wallet.rs or not. Step-by-step the handlers for the corresponding errors (e.g. `ClientError`, `InsufficientFunds`, ...) are added to this function. If you encounter an unhandled error, create the handler and add the type and the corresponding handlers in the following position in the `handleWalletRsError.ts` :
```

```typescript
switch (error?.type) {
    ...
    case WalletRsError.XXXXXXX:

>         handleXXXXXXXXError(error)

Reviewer: It looks nicer to use a concrete example instead of `XXXXXX`
